indico/modules/events/controllers/admin.py

```diff
@@ -149,3 +152,20 @@ def _process(self):
             flash(_('Settings have been saved'), 'success')
             return redirect(url_for('events.unlisted_events'))
         return WPEventAdmin.render_template('admin/unlisted_events.html', 'unlisted_events', form=form)
+
+
+class RHAutoLinker(RHAdminBase):
+    """Manage patterns which are automatically linked in notes."""
+
+    def _process(self):
+        return WPEventAdmin.render_template('admin/autolinker.html')
+
+
+class RHAutoLinkerConfig(RHAdminBase):
+    """Update configuration of auto-linker"""
+
+    @use_kwargs({
+        'rules': fields.Nested(AutoLinkerRuleSchema(many=True))
+    })
+    def _process(self, rules):
+        autolinker_settings.set('rules', AutoLinkerRuleSchema().dump(rules, many=True))
```

Comment:
Hmm having to dump again feels a bit strange. I guess this is because the field returns a compiled regex? Maybe cleaner for the field to do its magic but return the string pattern so we avoid this round-trip? I can't imagine any cases where you actually want to use the compiled regex after receiving it from the schema...

Reply:
The only downside with that approach is that we have to then compile the regex during form validation, to access its internal (groups, etc...) But it's not the end of the world either. I've changed it to work in that way.